### PR TITLE
feat(telemetry): collect deploy mode in telemetry

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: proto/telemetry.proto [hunk 1]
```diff
@@ -19,6 +19,16 @@ enum TelemetryNodeType {
   TELEMETRY_NODE_TYPE_COMPACTOR = 4;
 }
 
+// Coo
+enum TelemetryClusterType {
+    TELEMETRY_CLUSTER_TYPE_UNSPECIFIED = 0;
+    TELEMETRY_CLUSTER_TYPE_SINGLE_NODE = 1;
+    TELEMETRY_CLUSTER_TYPE_TEST = 2;
+    TELEMETRY_CLUSTER_TYPE_DOCKER_COMPOSE = 3;
+    TELEMETRY_CLUSTER_TYPE_KUBERNETES = 4;
```
[the remark this answers is not in the document]
Comment (tabVersion) [reply proto/telemetry.proto:28]: Cloud use `TELEMETRY_CLUSTER_TYPE_HOSTED`, setting by @arkbriar
see https://github.com/risingwavelabs/risingwave/blob/28ca6bb1cf87b969fb2b8194c45ba47854aedfb2/src/common/src/telemetry/mod.rs#L32

[the remark this answers is not in the document]
Comment (arkbriar) [reply proto/telemetry.proto:28]: Not set yet.

File: proto/telemetry.proto [hunk 2]
```diff
@@ -19,6 +19,17 @@ enum TelemetryNodeType {
...
 
+// Coo
+enum TelemetryClusterType {
+  TELEMETRY_CLUSTER_TYPE_UNSPECIFIED = 0;
+  TELEMETRY_CLUSTER_TYPE_SINGLE_NODE = 1;
+  TELEMETRY_CLUSTER_TYPE_TEST = 2;
+  TELEMETRY_CLUSTER_TYPE_DOCKER_COMPOSE = 3;
+  TELEMETRY_CLUSTER_TYPE_KUBERNETES = 4;
+  TELEMETRY_CLUSTER_TYPE_HOSTED = 5;
+  TELEMETRY_CLUSTER_TYPE_STANDALONE = 6;
```
Comment (fuyufjh) [inline proto/telemetry.proto:29]: These concepts are not orthogonal. For example, the CI/CD pipeline may run tests in standalone mode, making it confusing to different people.

I tend to use 2 separated dimensions

1. Is it a test or development instance? If yes, should be excluded from telemetry. I think our existing `TELEMETRY_ENABLED` is enough for this usage.
2. If it's a production cluster, how it is deployed? e.g. docker-compose, Kubernetes or standalone.

File: proto/telemetry.proto [hunk 3]
```diff
@@ -19,6 +19,16 @@ enum TelemetryNodeType {
   TELEMETRY_NODE_TYPE_COMPACTOR = 4;
 }
 
+enum TelemetryClusterType {
+  TELEMETRY_CLUSTER_TYPE_UNSPECIFIED = 0;
+  TELEMETRY_CLUSTER_TYPE_SINGLE_NODE = 1;
```
Comment (fuyufjh) [inline proto/telemetry.proto:24]: No need to have both `SINGLE_NODE` and `STANDALONE`.

- `risingwave single-node` or `risingwave` is the actual way to run a single-process RisingWave instance from operating system directly without container.
- `risingwave standalone` is only used in container (docker compose or Kubernetes), to run multiple roles in one container. In another word, the `TelemetryClusterType` should never be `STANDALONE`, but should be either `KUBERNETES` or `DOCKER_COMPOSE` instead.

File: proto/telemetry.proto [hunk 4]
```diff
@@ -19,6 +19,16 @@ enum TelemetryNodeType {
...
+  TELEMETRY_CLUSTER_TYPE_TEST = 2;
+  TELEMETRY_CLUSTER_TYPE_DOCKER_COMPOSE = 3;
+  TELEMETRY_CLUSTER_TYPE_KUBERNETES = 4;
+  TELEMETRY_CLUSTER_TYPE_HOSTED = 5;
```
Comment (fuyufjh) [inline proto/telemetry.proto:28]: What is `HOSTED` ?

Comment (tabVersion) [reply proto/telemetry.proto:28]: It means the cluster is on Risingwave Cloud.

Comment (fuyufjh) [reply proto/telemetry.proto:28]: I see. Perhaps you mean `CLOUD_HOSTED`

File: proto/telemetry.proto (hunk 2 above)
Comment (tabVersion) [reply proto/telemetry.proto:29]: Make sense, let's make the `TELEMETRY_ENABLED` default to true.

File: src/common/src/telemetry/mod.rs [hunk 5]
```diff
@@ -29,8 +29,48 @@ use crate::util::resource_util::memory::{system_memory_available_bytes, total_me
...
-const TELEMETRY_CLUSTER_TYPE_TEST: &str = "test"; // test environment, eg. CI & Risedev
+pub const TELEMETRY_CLUSTER_TYPE_HOSTED: &str = "hosted"; // hosted on RisingWave Cloud
+pub const TELEMETRY_CLUSTER_TYPE_TEST: &str = "test"; // test environment, eg. CI & Risedev
+pub const TELEMETRY_CLUSTER_TYPE_KUBERNETES: &str = "kubernetes";
+pub const TELEMETRY_CLUSTER_TYPE_SINGLE_NODE: &str = "single-node";
+pub const TELEMETRY_CLUSTER_TYPE_DOCKER_COMPOSE: &str = "docker-compose";
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum TelemetryClusterType {
+    Hosted,
+    Test,
```
Comment (fuyufjh) [inline src/common/src/telemetry/mod.rs:40]: Was not updated along with the proto. 

Why not just use the `enum TelemetryClusterType` in the generated code?

Comment (tabVersion) [reply src/common/src/telemetry/mod.rs:40]: Unfortunately, the 'test' have been introduced along with scarf pr. So keep for compatibility.

Comment (lmatz) [reply src/common/src/telemetry/mod.rs:40]: Does a breaking change of test affect us internal pipelines only or also users?

Comment (tabVersion) [reply src/common/src/telemetry/mod.rs:40]: Oh, the test pipeline does not enable telemetry, will remove later.

File: proto/telemetry.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (arkbriar) [reply proto/telemetry.proto]: Set now.

**Review by fuyufjh — APPROVED** [review]

**Pull request merged**